### PR TITLE
 Using Primary IP as inventory_hostname

### PR DESCRIPTION
**Pull request by aventrax**
Adding an option to use primary IP as inventory_hostname in place of the netbox device name.

### CODE REVIEW

**Comment by ryanmerolle** [comment]
@aventrax the CI failed on some simple black tests.  Feel free to peruse them and update accordingly

**Comment by aventrax** [comment]
@ryanmerolle I run black but it modifies even something not touched by me. What should I do? (row 451, it reformat "location", "rack_group", "services", "interfaces" and "dns_name").

**Review by aventrax — COMMENTED** [review]
I manually changed what seems to be the issue with indentation. Hope it works.

File: plugins/inventory/nb_inventory.py [hunk 1]
```diff
@@ -1578,7 +1583,11 @@ def main(self):
                 # Device is part of a virtual chassis, but is not the master
                 continue
 
-            hostname = self.extract_name(host=host)
+            if self.ansible_inventory_primary_ip and self.extract_primary_ip(host=host):
+                hostname = self.extract_primary_ip(host=host)
```
Comment (sc68cal) [inline plugins/inventory/nb_inventory.py:1587]: This is a little hacky. I don't know if I like setting a variable named `hostname` with a raw IP address in some cases.

**Comment by sc68cal** [comment]
I think I would prefer that this set the [`ansible_host`] setting for a host.

File: plugins/inventory/nb_inventory.py (hunk 1 above)
Comment (aventrax) [reply plugins/inventory/nb_inventory.py:1587]: I agree with you. Unfortunately Ansible itself has IP addresses straight on the inventory in their documentation, so any var used for that could contain an hostname or an IP.

Comment (sc68cal) [reply plugins/inventory/nb_inventory.py:1587]: Yes but my nit is that in most of the Ansible internals, they use `host` - which could either be an IP or a DNS name. You are assigning an IP address to a variable called `hostname` which, in my opinion, breaks the meaning of the variable.

**Pull request closed without merging**